include/Beman/Optional26/optional.hpp

```diff
@@ -950,209 +1106,466 @@ auto optional_map_impl(Opt&& opt, F&& f)
 template <class T>
 class optional<T&> {
```

Comment:
What do you think to split this file? Maybe:

* move `class optional<T>` into `details/optional_impl.hpp`
* move`class optional<T&>` into `details/optional_ref_impl.hpp`
* include both in `optional.hpp`

If we agree with such approach, I'm OK to defer the work into a new issue.